Il2CppInspector.Common/Reflection/TypeInfo.cs

```diff
@@ -294,6 +294,8 @@ public class TypeArgumentsComparer : EqualityComparer<TypeInfo[]>
             var index = name.IndexOf("`", StringComparison.Ordinal);
             if (index != -1)
                 name = name.Remove(index);
+            //get rid of compiler generate characters that's c# not support
+            name = name.Replace("<", "_").Replace(">", "_");
```

Comment:
This still changes `UnmangledBaseName` which is not a C#-specific name property. Since every use of `unmangleName` except for the definition of `UnmangledBaseName` is used for a C#-specific name, however, I think the best solution is to just change `UnmangledBaseName` so it doesn't use `unmangleName()` anymore.